main.tf

```diff
@@ -2,8 +2,8 @@ locals {
   # Atlantis
   atlantis_url = "https://${try(coalesce(
```

Comment:
I think this all can be re-written and simplified to:

```hcl
atlantis_url = "https://${try(
    var.atlantis.fqdn,
    module.alb.route53_records["A"].fqdn,
    module.alb.dns_name,
    "",
  )}"
```

Reply:
I tried that, but I got this error:
```
╷
│ Error: Invalid template interpolation value
│ 
│   on ../../main.tf line 3, in locals:
│    3:   atlantis_url = "https://${try(
│    4:     var.atlantis.fqdn,
│    5:     module.alb.route53_records["A"].fqdn,
│    6:     module.alb.dns_name,
│    7:     ""
│    8:   )}"
│     ├────────────────
│     │ module.alb.dns_name is null
│     │ module.alb.route53_records is object with no attributes
│     │ var.atlantis is object with no attributes
│ 
│ The expression result is null. Cannot include a null value in a string template.
╵
```

I think the problem is that `module.alb.dns_name` is always defined, even if it is `null`. Therefore, it doesn't throw an error, and `try` returns `null`.

I tried wrapping the `try` call in a call to `coalesce`, but that didn't fix the error, since `coalesce` raises an error when the final result is an empty string.

So I think we need the call to `coalesce` to drop the load balancer DNS name if it is null, and we need the `try` outside the `coalesce` to handle the case when there is no valid name. I removed the `try` around `module.alb.dns_name`, as it doesn't seem to be needed.